cast/src/lib.rs

```diff
@@ -447,8 +453,40 @@ impl SimpleCast {
         format!("0x{}", s)
     }
 
+    pub async fn generate_interface(
+        address_or_path: String,
+        chain: Chain,
+        etherscan_api_key: Option<String>,
+    ) -> Result<Vec<InterfaceSource>> {
+        let mut contract_names: Vec<String> = Vec::new();
+        let contract_abis: Vec<Abi> = if Path::new(&address_or_path).exists() {
```

Comment:
shouldn't be needed

```suggestion
        let contract_abis = if Path::new(&address_or_path).exists() {
```

I'd do this like below, instead of creating a mutable outside of the if/else

```rust
let (abis, names) = if ... {
     (x, y)
} else {
    (x, y)
};
```


If the path does not exist, I would error, instead of silently falling back to etherscan